app/js/Tab.js

```diff
@@ -26,7 +26,9 @@ class Tab {
    * @param {Element} container
    */
 
-  constructor(container) {
+  constructor(container, tabName) {
```

Comment:
I added title tag to the tab which will show the full filename. I think it is valid to have the same filename in two tabs. For example someone could have two files with the same name in different folders. Changing the display name could be confusing.